src/core/createEvent.js

```diff
@@ -10,28 +10,88 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
-import { clone, isEmptyObject, createMerger } from "../utils";
+import { isEmptyObject, deepAssign } from "../utils";
 
 export default () => {
   const content = {};
   let userXdm;
   let userData;
   let documentMayUnload = false;
-  let lastChanceCallback;
+  let isFinalized = false;
+  let shouldSendEvent = true;
 
   const event = {
+    getUserXdm() {
+      return userXdm;
+    },
     setUserXdm(value) {
+      if (isFinalized) {
```

Comment:
I wonder if we should extract those 3 lines into a quick helper, like `throwIfEventFinalized`..

Reply:
Done